rust-runtime/aws-smithy-http-server/src/response.rs

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+//! Serialize HTTP response and error traits
+
+/// `SerializeHttpResponse` is a generic trait for serializing structured data into HTTP responses.
+///
+/// It is designed to be flexible, because `Output` and `Struct` are unconstrained, it can be used to support
+/// event streams, regular request-response style operations, as well as any other HTTP-based
+/// protocol that we manage to come up with.
+///
+/// It also enables this critical and core trait to avoid being async, and it makes code that uses
+/// the trait easier to test.
+///
+/// TODO: streaming and not-fully loaded body serialization is no developed yet
+pub trait SerializeHttpResponse {
+    /// Struct instance to be serialized into the HTTP body
+    ///
+    /// For request/response style operations, this is typically something like:
+    /// `OperationOutput`
+    type Struct;
```

Comment:
Why isn't it this struct the one that directly `impl`s `SerializeHttpResponse`?

Reply:
Oh right, this is an excellent idea, I don't know why I did not think about it. I'll fix this.